### ISSUE TITLE
Parses invalid versions with leading `0` in numeric prerelease identifiers

### ISSUE THREAD

**Issue by Nemo157** [dtolnay/semver#206]
repro ([playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=7a14d152b0f8234a6ce0bbcb5a26ed78)):

```rust
fn main() {
    println!("{:#?}", semver::Version::parse("1.0.0-alpha.01"));
}
```

prints:

```
Ok(
    Version {
        major: 1,
        minor: 0,
        patch: 0,
        pre: [
            AlphaNumeric(
                "alpha",
            ),
            AlphaNumeric(
                "01",
            ),
        ],
        build: [],
    },
)
```

when it should be some form of error.

See https://github.com/semver/semver/issues/563 for clarification that this should be an error case, and cc downstream bug https://github.com/rust-lang/crates.io/issues/2463.

**Comment by steveklabnik** [comment]
Thanks, this is 100% a bug. I cannot guarantee that I will be patching it soon, but would accept a PR fixing this.

**Comment by dtolnay** [comment]
This is fixed in 1.0.0.

```rust
fn main() {
    println!("{}", semver::Version::parse("1.0.0-alpha.01").unwrap_err());
}
```

```console
invalid leading zero in pre-release identifier
```
